edgelet/hsm-sys/azure-iot-hsm-c/src/edge_pki_openssl.c

```diff
@@ -1881,3 +1882,38 @@ int verify_certificate
 
     return result;
 }
+
+int generate_rand_buffer(unsigned char *buffer, size_t num_bytes)
+{
+    int result;
+
+    initialize_openssl();
+
+    if ((buffer == NULL) || (num_bytes == 0) || (num_bytes > INT_MAX))
+    {
+        LOG_ERROR("Invalid parameters");
+        result = __FAILURE__;
+    }
+    else
+    {
+        // note RAND_bytes will seed the random number generator
+        // if not initialized once per process or unless cleaned up
+        // In other works we don't have to worry about seeding for
+        // the default implementation.
+        // https://wiki.openssl.org/index.php/Random_Numbers
+        int rc = RAND_bytes(buffer, (int)num_bytes);

```

Comment:
Docs say that this is deprecated in favor of `RAND_pseudo_bytes` in 1.1.0, and that `RAND_pseudo_bytes` is available since 0.9.5, so it should be available for all our supported distros?

Reply:
nvm, looking at [the actual source](https://github.com/openssl/openssl/blob/master/include/openssl/rand.h#L44) the situation is exactly backwards. The layout of the docs confused me.

Reply:
Yeah that got me too.